flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/metadata/FlinkRelMdWindowProperties.scala

```diff
@@ -92,13 +93,57 @@ class FlinkRelMdWindowProperties private extends MetadataHandler[FlinkMetadata.W
       return null
     }
 
+    val windowSpec = childProps.getWindowSpec
+    // if the window a
+    if (!validateWindowProperties(windowSpec, mapInToOutPos)) {
+      // If the window becomes illegal after passing through calc or project,
+      // return no window properties
+      return null
+    }
+
     childProps.copy(
       transformColumnIndex(childProps.getWindowStartColumns, mapInToOutPos),
       transformColumnIndex(childProps.getWindowEndColumns, mapInToOutPos),
-      transformColumnIndex(childProps.getWindowTimeColumns, mapInToOutPos)
+      transformColumnIndex(childProps.getWindowTimeColumns, mapInToOutPos),
+      updateWindowSpec(windowSpec, mapInToOutPos)
     )
   }
 
+  /**
+   * Validate the window properties when passing through calc or project.
+   *
+   * This method only checks the window that contains partition keys like session window. See more
```

Comment:
like => for?

Reply:
Update it to "the window like session window that contains partition keys"